workspaces/src/network/transaction.rs

```diff
@@ -0,0 +1,323 @@
+        Self {
+            client,
+            signer,
+            receiver_id,
+            actions: Vec::new(),
+        }
+    }
+
+    /// Adds a key to the `receiver_id`'s account, where the public key can be used
+    /// later to delete the same key.
+    pub fn add_key(mut self, pk: PublicKey, ak: AccessKey) -> Self {
+        self.actions.push(
+            AddKeyAction {
+                public_key: pk.into(),
+                access_key: ak.into(),
+            }
+            .into(),
+        );
+        self
+    }
+
+    /// Call into the `receiver_id`'s contract with the specific function arguments.
+    pub fn call(mut self, call_args: CallArgs) -> Self {
+        self.actions.push(call_args.into());
+        self
+    }
+
+    /// Create a new account with the account id being `receiver_id`.
+    pub fn create_account(mut self) -> Self {
+        self.actions.push(CreateAccountAction {}.into());
+        self
+    }
+
+    /// Deletes the `receiver_id`'s account. The beneficiary specified by
+    /// `beneficiary_id` will receive the funds of the account deleted.
+    pub fn delete_account(mut self, beneficiary_id: AccountId) -> Self {
+        self.actions
+            .push(DeleteAccountAction { beneficiary_id }.into());
+        self
+    }
+
+    /// Deletes a key from the `receiver_id`'s account, where the public key is
+    /// associated with the access key to be deleted.
+    pub fn delete_key(mut self, pk: PublicKey) -> Self {
+        self.actions
+            .push(DeleteKeyAction { public_key: pk.0 }.into());
+        self
+    }
+
+    /// Deploy contract code or WASM bytes to the `receiver_id`'s account.
+    pub fn deploy(mut self, code: Vec<u8>) -> Self {
+        self.actions.push(DeployContractAction { code }.into());
+        self
+    }
+
+    /// An action which stakes the signer's tokens and setups a validator public key.
+    pub fn stake(mut self, stake: Balance, pk: PublicKey) -> Self {
+        self.actions.push(
+            StakeAction {
+                stake,
+                public_key: pk.0,
+            }
+            .into(),
+        );
+        self
+    }
+
+    /// Transfer `deposit` amount from `signer`'s account into `receiver_id`'s account.
+    pub fn transfer(mut self, deposit: Balance) -> Self {
+        self.actions.push(TransferAction { deposit }.into());
+        self
+    }
+
+    async fn transact_raw(self) -> anyhow::Result<FinalExecutionOutcomeView> {
+        send_batch_tx_and_retry(self.client, &self.signer, &self.receiver_id, self.actions).await
+    }
+
+    /// Process the trannsaction, and return the result of the execution.
+    pub async fn transact(self) -> anyhow::Result<CallExecutionDetails> {
+        self.transact_raw().await.map(Into::into)
+    }
+}
+
+/// Similiar to a [`Transaction`], but more specific to making a call into a contract.
+/// Note, only one call can be made per `CallTransaction`.
+pub struct CallTransaction<'a, T> {
```

Comment:
I don't think it'll be the most intuitive to use for a single `call` function since they're not really the same APIs. Also would need to import `Function` every time now.

```rust
contract.call(
    worker,
    Function::new(name)
       .args_json(...)
)
.transact();
```

Which looks kinda funky over just:
```rust
contract.call(worker, name)
     .args_json(...)
     .transact()
```